lib/Disco.php

```diff
@@ -844,4 +874,43 @@ private static function constructSearchData($idpMetadata): string
 
         return strtolower(str_replace('"', '', iconv('UTF-8', 'US-ASCII//TRANSLIT', $res)));
     }
+
+    private static function substrInArray($needle, array $haystack)
+    {
+        foreach ($haystack as $item) {
+            if (strpos($item, $needle) !== false) {
+                return $item;
+            }
+        }
+
+        return null;
+    }
+
+    private function fillSpName($state)
+    {
+        $translate = new Translate(Configuration::getInstance());
```

Comment:
Please make use of the translation available in the template. Pass the template into the function and use it like `$t->t('{key}')`